sql/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -0,0 +1,268 @@
+        return extractMatches(columns, symbol, true);
+    }
+
+    private List<List<Symbol>> extractMatches(Collection<ColumnIdent> columns, Symbol symbol, boolean exact){
+        EqualityExtractor.ProxyInjectingVisitor piv = new EqualityExtractor.ProxyInjectingVisitor();
+        EqualityExtractor.ProxyInjectingVisitor.Context context =
+                new EqualityExtractor.ProxyInjectingVisitor.Context(columns, exact);
+        Symbol proxiedTree = piv.process(symbol, context);
+
+        // bail out if we have any unknown part in the tree
+        if (context.exact && context.seenUnknown){
+            return null;
+        }
+
+        //proxiedTree = normalizer.normalize(proxiedTree);
+        List<Set<EqProxy>> comparisons = context.comparisonSet();
+
+
+        Set<List<EqProxy>> cp = Sets.cartesianProduct(comparisons);
+
+        List<List<Symbol>> result = new ArrayList<>();
+        for (List<EqProxy> proxies : cp) {
+            boolean anyNull = false;
+            for (EqProxy proxy : proxies) {
+                if (proxy != NULL_MARKER_PROXY){
+                    proxy.setTrue();
+                } else {
+                    anyNull = true;
+                }
+            }
+            Symbol normalized = normalizer.normalize(proxiedTree);
+            if (normalized == Literal.BOOLEAN_TRUE){
+                if (anyNull){
+                    return null;
+                }
+                List<Symbol> row = new ArrayList<>(proxies.size());
+                for (EqProxy proxy : proxies) {
+                    proxy.reset();
+                    row.add(proxy.origin.arguments().get(1));
+                }
+                result.add(row);
+            } else {
+                for (EqProxy proxy : proxies) {
+                    proxy.reset();
+                }
+            }
+        }
+        return result.isEmpty() ? null : result;
+
+    }
+
+    static class EqProxy extends Symbol {
+
+        private Symbol current;
+        private final Function origin;
+
+        public Function origin() {
+            return origin;
+        }
+
+        EqProxy(Function origin) {
+            this.origin = origin;
+            this.current = origin;
+        }
+
+        public void reset(){
+            this.current = origin;
+        }
+
+        public void setTrue(){
+            this.current = Literal.BOOLEAN_TRUE;
+        }
+
+        @Override
+        public SymbolType symbolType() {
+            return current.symbolType();
+        }
+
+        @Override
+        public <C, R> R accept(SymbolVisitor<C, R> visitor, C context) {
+            return current.accept(visitor, context);
+        }
+
+        @Override
+        public DataType valueType() {
+            return current.valueType();
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            throw new NotImplementedException();
```

Comment:
use UnsupportedOperation instead. NotImplementedException results in warnings (internal proprietary API)